src/Raven.Server/Web/Studio/Sharding/Processors/ShardedBucketsHandlerProcessorForGetBuckets.cs

```diff
@@ -0,0 +1,90 @@
+﻿using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using JetBrains.Annotations;
+using Microsoft.AspNetCore.Http;
+using Raven.Client.Http;
+using Raven.Server.Documents.Sharding.Executors;
+using Raven.Server.Documents.Sharding.Handlers;
+using Raven.Server.Documents.Sharding.Operations;
+using Raven.Server.ServerWide.Context;
+using Raven.Server.Web.Studio.Processors;
+
+namespace Raven.Server.Web.Studio.Sharding.Processors
+{
+    internal class ShardedBucketsHandlerProcessorForGetBuckets : AbstractBucketsHandlerProcessorForGetBuckets<ShardedDatabaseRequestHandler, TransactionOperationContext>
+    {
+        public ShardedBucketsHandlerProcessorForGetBuckets([NotNull] ShardedDatabaseRequestHandler requestHandler) : base(requestHandler)
+        {
+        }
+        
+        protected override async ValueTask<BucketsResults> GetBucketsResults(TransactionOperationContext context, int fromBucket, int toBucket, int range, CancellationToken token)
+        {
+            var shardedGetBucketsOperation = new ShardedGetBucketsOperation(RequestHandler.HttpContext.Request, fromBucket, toBucket, range);
+            return await RequestHandler.ShardExecutor.ExecuteParallelForAllAsync(shardedGetBucketsOperation, token);
+        }
+    }
+
+    public readonly struct ShardedGetBucketsOperation : IShardedOperation<BucketsResults>
+    {
+        private readonly HttpRequest _request;
+        private readonly int _fromBucket;
+        private readonly int _toBucket;
+        private readonly int _range;
+
+        public ShardedGetBucketsOperation(HttpRequest httpRequest, int fromBucket, int toBucket, int range)
+        {
+            _request = httpRequest;
+            _fromBucket = fromBucket;
+            _toBucket = toBucket;
+            _range = range;
+        }
+
+        public HttpRequest HttpRequest => _request;
+
+        public BucketsResults Combine(Dictionary<int, ShardExecutionResult<BucketsResults>> results)
+        {
+            var bucketRanges = new BucketsResults();
+
+            foreach (var shardResult in results.Values)
+            {
+                var shardStats = shardResult.Result;
+
+                bucketRanges.TotalSize += shardStats.TotalSize;
+                
+                foreach (var (range, cur) in shardStats.BucketRanges)
```

Comment:
what are we aggregating here? and why?
or we are not going to show only local buckets any more?
but will show something like
```
Buckets [0..1024]
- Total Size 100 MB
- Total Docs 50
- Shards [1,3,4]
```

Reply:
Yes, that is what we'll do.
The idea is that there is a single endpoint, at the shard level, which you can ask to get bucket ranges, all the way down to a single bucket.
Gives us a simpler API for the Studio, I believe.